### DISCUSSION TITLE
Pull Request Templates improvements: support `.github-private` and UI for picking template

### DISCUSSION THREAD

**Discussion by Ignigena** [community/community#42499] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

The ability to create organization-wide Pull Request templates is fantastic -- but currently requires placing the template in a public `.github` repository. It does not seem to be supported with the new `.github-private` repository which can be used for members-only profile contents.

It would be fantastic to have the option to store a PR template in the private repository which is applied for any Private repos on the organization with a similar reasoning to the members-only profile snippet: sometimes we want a specific PR template to show for internal contributors that are different from the one shown on public repositories. Linking to internal pages and other such customizations.

Also, it appears that one can create multiple PR templates by creating one or more markdown files at `PULL_REQUEST_TEMPLATE/*.md` which is fantastic -- having separate templates for bug fixes vs. features for example would be great. Have used this quite frequently on the Issue templates and one of the biggest values of this is the UI that it enables which forces a user to pick the issue type and see the appropriate template ex: https://github.com/angular/angular/issues/new/choose

However, when one creates multiple pull request templates what actually happens is that _none_ of the templates are used by default unless a special `?template=` query param is added the URL. This makes the PR templates a bit less useful since it requires that users follow a specific link while creating the PR vs. being prompted to choose via the GitHub UI on the Web.

Lastly ... can we please have issue form YAML syntax for PR templates 🙏🏻 

That's all for now 😄

**Comment by ARolek** [comment]
I'm also encountering this situation. This seems like a great use of the special `.github-private` repo.

**Comment by james-trousdale-lyb** [comment]
:+1: - our company has migrated to an EMU instance, and this is a major feature gap right now as nothing public can exist in the enterprise.

**Comment by theprivileges** [comment]
Support for `PULL_REQUEST_TEMPLATE.md` in `.github-private` would be 🔥

**Comment by Spelkington** [comment]
A vocal +1 for `.github-private` behaving as... well, `.github` but private. The only mention of `.github-private` in the documentation is [a passing mention of support for private `README.md`](https://github.com/github/docs/blob/98510f795bc25ff6acf56f7951492ab1d6acfb53/content/organizations/collaborating-with-groups-in-organizations/customizing-your-organizations-profile.md?plain=1#L56) - I'd be shocked if this is where support for the `.github-private` repo stops.

**Comment by matthewtusker** [comment]
I'm also very keen to see this being implemented. My assumption until I started looking into it would have been that `.github-private` would act like `.github`, but only apply to private repositories. It seems that this isn't the case.

**Comment by acrogenesis** [comment]
This is still an issue

**Comment by brentsnook** [comment]
We're also looking for a way to display a public README for the org while still keeping our PR templates private. The proposed solution of supporting PR templates in `.github-private` would work perfectly for us - as it stands now I can't see another way to achieve this.

**Comment by nittendev** [comment]
This is very much still an issue.

**Comment by dudo** [comment]
This does seem like a nice improvement, but realistically speaking, what's sensitive about issue/pr/discussion templates that you'd want to hide from the public? I actually like having them public as a way to show how tight our process is.

**Reply by Spelkington** [reply]
Having worked at a combination of organizations requiring high compliance and low compliance, I think the issue isn't that issue/pr/discussion templates are critical data, but instead that satisfying compliance requirements is typically done in broad strokes, taking a better-safe-than-sorry approach to minimize any risk of internal information leaks.

Unfortunately, the current system creates a circumstance where satisfying internal organization best practices and internal compliance best practices are a strict trade-off, when they probably don't need to be.

**Reply by james-trousdale-lyb** [reply]
Moreover, this advice ignores that [GitHub themselves provide a product](https://docs.github.com/en/enterprise-cloud@latest/admin/identity-and-access-management/understanding-iam-for-enterprises/about-enterprise-managed-users) wherein it's impossible to have a public `.github` repo.

**Reply by zrosenbauer** [reply]
Its not even wanting them to be private or public... I have very different requirements for an OSS project vs our own internal project, i.e. adding a "checkmark" for "QA team has signed off on staging" or something even just "Submit Security Vulnerabilities to our OSS Form vs our Internal Form"

**Reply by dipakparmar** [reply]
Similarly, we have some links to tools and documents, specific checks that we prefer to not be public for compliance reasons.

**Reply by MichaelBonnet** [reply]
I work in aerospace. Having anything public runs compliance and legal risk.

**Reply by deiga** [reply]
Luckily they finally made private `.github` repos a possibility! But only for `workflow-templates` currently https://github.blog/changelog/2025-09-18-actions-yaml-anchors-and-non-public-workflow-templates/#use-workflow-templates-from-non-public-github-repositories

**Comment by hyunsu-kang** [comment]
This has been an ongoing issue and is a feature that absolutely needs to be added.
There is a recently discovered security case. 
"Hackers are using unpublished GitHub and GitLab comments to generate phishing links that appear to come from legitimate open source software (OSS) projects."
source: https://www.darkreading.com/threat-intelligence/hackers-create-legit-phishing-links-with-ghost-github-gitlab-comments

Even because of these cases, this is a feature that must be added.

**Reply by 0otafasd** [reply]
+1

**Reply by levtomer66** [reply]
+1

**Reply by tscholz-hi** [reply]
+1

**Comment by ndlanier** [comment]
I'd rather not have to maintain pr templates for each repo across an org to keep a good security posture. These seems like an extremely necessary feature to allow for privated/internal .github repositories.

**Comment by jsebringStrata** [comment]
+1

**Comment by nikhilm-yai** [comment]
+1

**Comment by aalkayfeactive** [comment]
+1

**Comment by nc-tpi** [comment]
+1

**Comment by jiaranda** [comment]
+1

**Comment by etbutler** [comment]
+1

**Comment by cdco-paul-legrand** [comment]
+1

**Comment by zade107** [comment]
+1

**Comment by margarita-zabelina-cko** [comment]
+1 pls

**Comment by SameerKhan122** [comment]
+1

**Comment by metalfacesec** [comment]
+1

**Comment by samuelshu-ubnt** [comment]
+1

**Comment by stroebel** [comment]
+1

**Comment by geocomm-ascuccimarra** [comment]
+1

**Comment by elazo89** [comment]
Is there any hope to have this feature in Pull Request soon? I really wish to have this working.

**Comment by nivrith-jr** [comment]
+ 1

**Comment by Piidgey** [comment]
+ 1
Especially for the part about the url query param (?template=). This just don't make any sense.

**Comment by ItsIgnacioPortal** [comment]
Please stop posting "+1" and instead click the upvote icon on the original post. Posting "+1" does not give this issue more attention, it only sends a spammy notification to all 34 people that are subscribed to this issue.
<br>
![image](https://github.com/user-attachments/assets/9e9da7de-8fd4-401b-9ab8-0c73510a08ac)

**Reply by addisonlynch** [reply]
+1

**Comment by addisonlynch** [comment]
+1

**Comment by 4Source** [comment]
> Lastly ... can we please have issue form YAML syntax for PR templates 🙏🏻

I am also waiting for this since issue templates with YAML where introduced but I doesn't seem like this is planned...

> However, when one creates multiple pull request templates what actually happens is that _none_ of the templates are used by default unless a special `?template=` query param is added the URL. This makes the PR templates a bit less useful since it requires that users follow a specific link while creating the PR vs. being prompted to choose via the GitHub UI on the Web.

I think this would be necessary **if** YAML PR templates will be introduced

**Comment by BlckHawker** [comment]
+1

**Comment by agneszitte** [comment]
+1

**Comment by yoavfeld** [comment]
+1

**Comment by kaydenpham27** [comment]
Does anyone have an elegant workaround to use private PR templates in query parameter?
